include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -4048,6 +4048,26 @@ DoFCellAccessor<DoFHandlerType, level_dof_access>::set_active_fe_index(
 
 
 
+template <typename DoFHandlerType, bool level_dof_access>
+inline const FiniteElement<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension> &
+DoFCellAccessor<DoFHandlerType, level_dof_access>::get_future_fe() const
+{
+  Assert(
+    (dynamic_cast<const dealii::DoFHandler<DoFHandlerType::dimension,
+                                           DoFHandlerType::space_dimension> *>(
+       this->dof_handler) != nullptr) ||
+      (this->has_children() == false),
+    ExcMessage("In hp::DoFHandler objects, finite elements are only associated "
+               "with active cells. Consequently, you can not ask for the "
+               "future finite element on cells with children."));
+  Assert(this->dof_handler != nullptr, typename BaseClass::ExcInvalidObject());
```

Comment:
I think you ought to move this check above the previous one. You'd probably get funny error messages otherwise where the text doesn't match the expected condition.

Reply:
Agree! We should do the same for the `get_fe` function, too.